### PR TITLE
Refactored device and variant selection

### PR DESCRIPTION
**Pull request by themarpe**
Adds description for selected device, and subsequently adds description for selected variant. 

This pins down the actual specific board in question, for better (less error-prone) UX.

image

TODO:
- [x] Port over all files and group accordingly
- [ ] Add support for images
- [x] Add support for automatically using the test_type specified by the "selected device"

### CODE REVIEW

**Comment by saching13** [comment]
Additional TODO(May be different PR):
 
1. Deciding on skipping NOR Flash if the device doesn't have one From EEPROM Config or any other json variable
2. Configure pipeline, Image resolution from Hardware Config in EEPROM json
3. Skip IMU Update based on Hardware config if the device doesn't have IMU.

**Comment by daxoft** [comment]
LGTM, I tested it with an oak-lite and it works, I'll add the other jsons, try some other boards, and I can add the images if needed.

File: batch/oak_1.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        {
+            "title":"OAK-1 Lite C05 (NG9093 R2M0E3)",
+            "description": "OAK-1 Lite C05",
+            "eeprom":"eeprom/NG9093_R2M0E3_oak_1_lite_c05.json"
+        },
+        {
+            "title":"OAK-1 Lite W (NG9093 R2M0E3)",
+            "description": "OAK-1 Lite Wide Lens",
+            "eeprom":"eeprom/NG9093_R2M0E3_oak_1_lite_w.json"
+        },
```
[the remark this answers is not in the document]
Comment (themarpe) [reply batch/oak_1.json:31]: @daxoft

File: test_ui.py [hunk 2]
```diff
@@ -642,7 +693,8 @@ def setupUi(self):
         # self.save_but.setObjectName("connect_but")
         # self.save_but.clicked.connect(save_csv)
         self.automated_tests = QtWidgets.QGroupBox(self.centralwidget)
-        if test_type == 'OAK-1':
+        # if test_type == 'OAK-1':
+        if 'oak-1' in eepromDataJson['productName'].lower():
```
[the remark this answers is not in the document]
Comment (themarpe) [reply test_ui.py:697]: @daxoft

[the remark this answers is not in the document]
Comment (daxoft) [reply test_ui.py:697]: @themarpe the 'test_type' is an artifact from when the board type was typed in the terminal, for now it's always OAK-D-PRO-Poe, will be removed in the future. Or, you would prefer to save something else into that and use it? Like device description, or the product name.

File: batch/oak_1.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (daxoft) [reply batch/oak_1.json:31]: ok, I'll move them to OAK-D Lite. Just though they are not "OAK-D"s

File: test_ui.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (themarpe) [reply test_ui.py:697]: Well, that could be the main way of specifying what kind of tests are ran. 
Eg OAK-D Pro PoE vs OAK-D Lite - different resolution/cameras/connectivity options, IR & Laser vs None, etc...

First prefer higher up abstraction with this kind of higher level enums that specify what things are run as part of test. Or maybe even do it "component based" where the device/variant downselects the list of tests of functionality. 

But rather not use eeprom json product name to specify that. Make it explicit. Product name can be anything, its an alias. Plus its a string - someone might type it differently and the whole thing will blow up.

[the remark this answers is not in the document]
Comment (daxoft) [reply test_ui.py]: ok, I switched to using the test_type

**Pull request merged**